File: Intervals/452. Minimum Number of Arrows to Burst Balloons/solution.ts
function findMinArrowShots(points: number[][]): number {
    if (!points.length) {
        return 0;
    }

    points.sort((a, b) => a[1] - b[1]);
    let arrows = 1;
    let curr = points[0];

    for (let balloon of points) {
        if (curr[1] < balloon[0]) {
            curr = balloon;
            arrows++;
        }
    }

    return arrows;
};

File: Intervals/452. Minimum Number of Arrows to Burst Balloons/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points:
            return 0

        points.sort(key=lambda x: x[1])
        arrows = 1
        curr = points[0]

        for balloon in points:
            if curr[1] < balloon[0]:
                curr = balloon
                arrows += 1

        return arrows
